lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -299,6 +308,64 @@ export abstract class ClientApplication {
         }
     }
 
+    /**
+     * This function redeems an authorization code (passed as code) from the eSTS token endpoint.
+     * This authorization code should be acquired server-side using a confidential client to acquire a spa_code.
+     * This API is not indended for normal authorization code acquisition and redemption.
+     * 
+     * Redemption of this authorization code will not require PKCE, as it was acquired by a confidential client.
+     * 
+     * @param request {@link AuthorizationCodeRequest}
+     * @returns A promise that is fulfilled when this function has completed, or rejected if an error was raised.
+     */
+    async acquireTokenByCode(request: AuthorizationCodeRequest): Promise<AuthenticationResult> {
+        this.preflightBrowserEnvironmentCheck(InteractionType.Silent);
+        this.logger.trace("acquireTokenByCode called", request.correlationId);
+        this.eventHandler.emitEvent(EventType.ACQUIRE_TOKEN_BY_CODE_START, InteractionType.Silent, request);
+
+        try {
+            if (!request.code) {
+                throw BrowserAuthError.createAuthCodeRequiredError();
+            }
+
+            let response = this.hybridAuthCodeResponses.get(request.code);
```

Comment:
Is there a specific time limit on the code validity? And in case if the code isn't redeemed on time, we throw a server error I assume?

Reply:
Auth codes have a short lifetime (5 minutes, I believe). This logic is to memoize the response, so that `acquireTokenByCode` is called multiple times in parallel, only one network request is made (and this gets cleared once the request has completed). And yes, if an app tries to redeem an auth code multiple times or after it has expired, an error will be returned from the server.